pretty_test.go

```diff
@@ -0,0 +1,123 @@
+package microerror
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestPretty(t *testing.T) {
+	testCases := []struct {
+		name            string
+		errorFactory    func() error
+		expectedMessage string
+	}{
+		{
+			name: "case 0: simple error",
+			errorFactory: func() error {
+				err := errors.New("something went wrong")
+
+				return err
+			},
+			expectedMessage: "Error: Something went wrong",
+		},
+		{
+			name: "case 1: simple empty error",
+			errorFactory: func() error {
+				err := errors.New("")
+
+				return err
+			},
+			expectedMessage: "",
+		},
+		{
+			name: "case 2: simple error with 'error:' prefix in message",
+			errorFactory: func() error {
+				err := errors.New("error: something went wrong")
+
+				return err
+			},
+			expectedMessage: "Error: Something went wrong",
+		},
+		{
+			name: "case 3: masked simple error",
+			errorFactory: func() error {
+				err := errors.New("something went wrong")
+
+				return Mask(err)
+			},
+			expectedMessage: "Error: Something went wrong",
+		},
+		{
+			name: "case 4: microerror, 0 depth",
+			errorFactory: func() error {
+				err := &Error{
+					Kind: "somethingWentWrongError",
+				}
+
+				return err
+			},
+			expectedMessage: "Error: Something went wrong",
+		},
+		{
+			name: "case 5: microerror, 1 depth",
+			errorFactory: func() error {
+				err := &Error{
+					Kind: "somethingWentWrongError",
+				}
+
+				return Mask(err)
+			},
+			expectedMessage: "Error: Something went wrong",
+		},
+		{
+			name: "case 6: microerror, 1 depth, with annotation",
+			errorFactory: func() error {
+				err := &Error{
+					Kind: "somethingWentWrongError",
+				}
+
+				return Maskf(err, "something bad happened, and we had to crash")
+			},
+			expectedMessage: "Error: Something bad happened, and we had to crash",
```

Comment:
Shouldn't it be `Error: Something went wrong: Something bad happened, and we had to crash`?

I'm thinking about `notFoundError` kind for example. We have code like: `return "", microerror.Maskf(notFoundError, "control plane CR")`

Then the error here would be:

```
Error: Control plane CR
```

I'd rather see:

```
Error: Not Found: Control plane CR
```

Reply:
`Error: Not Found: Control plane CR`
This doesn't look very good to me, but indeed it's an use case that we should support. I'll do it in this way, and leave room for improvement for later